src/isomorphic/classic/element/ReactElement.js

```diff
@@ -231,13 +223,30 @@ ReactElement.createElement = function(type, config, children) {
   // Children can be more than one argument, and those are transferred onto
   // the newly allocated props object.
   var childrenLength = arguments.length - 2;
+  var childArray;
   if (childrenLength === 1) {
-    props.children = children;
+    if (Array.isArray(children)) {
+      childArray = children.slice(0);
```

Comment:
Does this mean that if we pass them through multiple levels of components, the same array could get `slice(0)`d many times?